ports/esp32/modmachine.c

```diff
@@ -146,7 +148,7 @@ static void machine_sleep_helper(wake_type_t wake_type, size_t n_args, const mp_
         esp_sleep_enable_timer_wakeup(((uint64_t)expiry) * 1000);
     }
 
-    #if !CONFIG_IDF_TARGET_ESP32C3
+    #if !(CONFIG_IDF_TARGET_ESP32C3 || CONFIG_IDF_TARGET_ESP32C6)
```

Comment:
There are capability macros for esp_sleep APIs too, but they're granular

SOC_PM_SUPPORT_EXT0_WAKEUP, SOC_PM_SUPPORT_EXT1_WAKEUP, SOC_TOUCH_SENSOR_SUPPORTED, SOC_ULP_SUPPORTED

SOC_PM_SUPPORT_EXT1_WAKEUP and SOC_ULP_SUPPORTED are actually set on C6 but not C3, so some of these functions might work on C6

Not sure how keen you are to dig into the weeds here, though?

Reply:
Thanks for the suggestions, it's not really going into the weeds much for me now because you've done the work digging up these definitions!  I'm happy to put these into the code, though I don't really have hardware ready to test them all... Though these really feel like low risk charges.